File: Go/Arrays/average.go
package main

import "fmt"

func main() {
	scores := []int{72, 73, 33}

	fmt.Printf("Array length: %d\n", len(scores))

	var average = array_average(scores)
	fmt.Printf("Average: %f\n", average)
}

func array_average(scores []int) float32 {
	var total = 0 // type can be inferred

	for i := 0; i < len(scores); i++ {
		total += scores[i]
		fmt.Printf("Total: %d, %d\n", total, scores[i])
	}

	fmt.Printf("Total: %d\n", total)

	// have to cast both vars?
	return float32(total) / float32(len(scores))
}

File: Python/Arrays/average.py
# python3 average.py


def array_average(scores):
    total = 0

    for score in scores:
        total += score
        print(f"Total: {total}, {score}")

    print(f"Total: {total}")
    return total / len(scores)


scores = [72, 73, 33]
print(f"Array length: {len(scores)}")

average = array_average(scores)

print(f"Average: {average}")
